aiidalab_widgets_base/display.py

```diff
@@ -0,0 +1,22 @@
+from __future__ import print_function
+
+import importlib
+from IPython.display import display
+
+RECOGNIZED_AIIDA_DATA_VISUALIZERS = {
+    'data.parameter.ParameterData.' : 'ParameterDataVisualizer',
+    'data.structure.StructureData.' : 'StructureDataVisualizer',
+    'data.cif.CifData.'             : 'StructureDataVisualizer',
+    'data.folder.FolderData.'       : 'FolderDataVisualizer',
+    'data.array.bands.BandsData.'   : 'BandsDataVisualizer',
+}
+
+def aiidalab_display(obj, downloadable=True):
+    """Function that is able to display properly differnt AiiDA data types"""
+    from aiidalab_widgets_base import aiida_visualizers
+    if hasattr(obj, 'type') and obj.type in RECOGNIZED_AIIDA_DATA_VISUALIZERS:
```

Comment:
The more pythonic way would be to just try:

```python
try:
    visualizer = getattr(aiida_visualizers, AIIDA_DATA_VISUALIZER_MAP[obj.type])
    display( visualizer(obj, downloadable=downloadable), **kwargs)
except KeyError:
    display(obj, **kwargs)
```

Reply:
Really?

I am of course happy to change but are you sure `try-except` is any better in this case?

Reply:
Well, this is definitely shorter and more clear. I go for your advice.

Reply:
> shorter and more clear

That was also my thinking, following the "EAFP" motto.

Even if there might be a slight penalty (not sure there is), you are just trying once per visualizer so there is no point in worrying about it.